code/c++/search_letor.cc

```diff
@@ -0,0 +1,94 @@
+#include <xapian-letor.h>
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+using namespace std;
+
+static void show_usage()
+{
+    cout << "Usage: rank_letor --db=DIRECTORY MODEL_METADATA_KEY QUERY\n";
+}
+
+// Start of example code.
+// Stopwords:
+static const char * sw[] = {
+    "a", "about", "an", "and", "are", "as", "at",
+    "be", "by",
+    "en",
+    "for", "from",
+    "how",
+    "i", "in", "is", "it",
+    "of", "on", "or",
+    "that", "the", "this", "to",
+    "was", "what", "when", "where", "which", "who", "why", "will", "with"
+};
```

Comment:
IIRC, James said **NOT** to copy the example code from xapian-letor.

But if you're going to ignore that, you need to simplify it by removing stuff which doesn't illustrate what the documentation is trying to explain.

For example, this stopword stuff just complicates the code such that it's harder to follow the letor part.  Similarly, spelling correction just gets in the way here.